net-libs/libssh/libssh-0.8.4.recipe

```diff
@@ -57,43 +58,63 @@ BUILD_PREREQUIRES="
 	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
 	cmd:ld$secondaryArchSuffix
-	cmd:libtoolize
 	cmd:make
 	cmd:pkg_config$secondaryArchSuffix
 	"
 
+if [ -z "$secondaryArchSuffix" ]; then
+	ARCHITECTURES_doc="any"
+
+	PROVIDES_doc="
+		libssh_doc = $portVersion
+		"
+
+	BUILD_PREREQUIRES+="
+		cmd:doxygen
+		"
+fi
+
 defineDebugInfoPackage libssh$secondaryArchSuffix \
 	"$libDir"/libssh.so.$libVersion
 
 PATCH()
 {
-	# keyfile torture test is currently broken
+	# Requires: https://review.haiku-os.org/c/haiku/+/632
 	sed -i \
 		-e '/torture_threads_pki_rsa/d' \
 		-e '/torture_rand/d' \
 		-e '/torture_threads_init/d' \
 		-e '/torture_threads_buffer/d' \
 		-e '/torture_threads_crypto/d' \
-		tests/unittests/CMakeLists.txt || die
+		tests/unittests/CMakeLists.txt
+
+	# Doesn't work in chroot, probably due to the lack of the passwd database
+	sed -i 's/cmocka_unit_test(torture_path_expand_tilde_unix),//' \
+		tests/unittests/torture_misc.c
 }
 
 BUILD()
 {
+	_maybe_pie=()
+	if [ "$targetArchitecture" != x86_gcc2 ]; then
```

Comment:
@alaviss, I tried to push this tiny change, but I got a `(permission denied)` (probably because you did not check "Allow edits from maintainers.") That's OK. I also always uncheck it on my PRs :)
However, could you please make the following change before merging?
```diff
  	_maybe_pie=()
-	if [ "$targetArchitecture" != x86_gcc2 ]; then
+	if [ "$effectiveTargetArchitecture" = x86_64 ]; then
 		_maybe_pie=("-DCMAKE_EXE_LINKER_FLAGS=-pie -fstack-protector")
 	fi
```
This would allow the recipe to be 100% correct for those users who might choose to build a x86 primary arch system.
Also, when we test the CPU for a particular arch to decide which compiler options to use we should always use "$effectiveTargetArchitecture".
Thanks!